controllers/backend/volume/volume.go

```diff
@@ -55,3 +57,43 @@ func (v *volumeController) sync(key string, pvc *v1.PersistentVolumeClaim) error
 	_, err = v.volumes.Update(newVol)
 	return err
 }
+
+func (v *volumeController) Create(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	if pvc == nil {
+		return pvc, nil
+	}
+
+	if pvc.Labels["rio.cattle.io/use-templates"] == "" {
+		return pvc, nil
+	}
+
+	ns := pvc.Namespace
+	serviceName := pvc.Labels["app"]
+	templateName := strings.SplitN(pvc.Name, "-"+serviceName+"-", 2)
+	volTemplate, err := v.volumeLister.Get(ns, templateName[0])
+	if errors.IsNotFound(err) {
+		return pvc, nil
+	} else if err != nil {
+		return pvc, err
+	}
+
+	volume := &v1beta1.Volume{}
+	volume.Spec = volTemplate.Spec
+	volume.Name = pvc.Name
+	volume.Namespace = pvc.Namespace
+	volume.Spec.Template = false
+	volume.Labels = make(map[string]string)
+	volume.Labels["rio.cattle.io/volume-template"] = volTemplate.Name
```

Comment:
we have both "rio.cattle.io/volume-template" and "rio.cattle.io/volumetemplate" now.  Can we just change everything to be "rio.cattle.io/volume-template"  (the hyphen is more consistent with other labels)